### ISSUE TITLE
[Suggestion] Add AMD card compatibility

### ISSUE THREAD

**Issue by Ryanz48** [SociallyIneptWeeb/AICoverGen#55]

**Comment by SociallyIneptWeeb** [comment]
https://github.com/SociallyIneptWeeb/AICoverGen/issues/19

You can refer to this

**Comment by Ryanz48** [comment]
Yeah sorry, I didn't notice that issue. But the thing is, until yesterday the Colab notebook was my main option, but Google has started to crack down on the notebooks using RVC (not banning the notebooks, but users get disconnected even 5-10 minutes after starting a runtime, even banning some users. I think you should consider adding AMD compatibility because of this, or some kind of option that allows the CPU to do the inference.

**Comment by SociallyIneptWeeb** [comment]
Unfortunately, I do not have an AMD graphics card, so I can't really develop and test this properly on my environment.

**Comment by MidnightKittenCat** [comment]
I second this.

From the looks of it, it seems to be using pytorch so perhaps the rocm version of pytorch could add compatibility? I'll take a look.

**Comment by MidnightKittenCat** [comment]
![image](https://github.com/SociallyIneptWeeb/AICoverGen/assets/66114416/74527eb3-4ffa-41c6-8057-ee56f2c3b2e3)
The first part uses the cpu(splitting the vocals etc), afterwards it uses the gpu if used with pytorch rocm

**Comment by ayaldev** [comment]
> ![image](https://user-images.githubusercontent.com/66114416/269071245-74527eb3-4ffa-41c6-8057-ee56f2c3b2e3.png) The first part uses the cpu(splitting the vocals etc), afterwards it uses the gpu if used with pytorch rocm

i have rx 6800, how did you do this?

**Comment by Denis6902** [comment]
> ![image](https://user-images.githubusercontent.com/66114416/269071245-74527eb3-4ffa-41c6-8057-ee56f2c3b2e3.png) The first part uses the cpu(splitting the vocals etc), afterwards it uses the gpu if used with pytorch rocm

is it possible to do this with directml ?

**Comment by YourSandwich** [comment]
For me on Linux with ROCm 5.6 and torch 2.1.0 it crashes with SIGSEV

I have adjusted `CUDAExecutionProvider` with `ROCMExecutionProvider` bot no change, it crashes at the first section after pressing Generate.

I am willing to help to bring support for AMD, unfortunately my Graphics card RX 7900XT requires ROCm 5.6 and torch2.1 to work.  

I am able to use this Project without any issues: https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/tree/main

It uses the same UVR5 models etc., so I don't understand, but even with a working VENV from that project it still does not work here.

That was a lot of I here 😰

**Comment by YourSandwich** [comment]
NVM i forgot that i have two graphics cards, fixed it with
export PYTORCH_ROCM_ARCH="gfx1100"
export ROCR_VISIBLE_DEVICES=0


here is the installation process:

```bash
python3.10 -m venv venv
source venv/bin/activate
pip install --pre torch torchaudio --index-url https://download.pytorch.org/whl/rocm5.6

# Remove torch from requirements.txt
pip install -r requirements.txt

export PYTORCH_ROCM_ARCH="gfx1100" # Probably not needed for you, also adjust the model
export ROCR_VISIBLE_DEVICES=0                # Probably not needed for you
python src/download_models.py
python src/webui.py
```

If this does not work also replace `CUDAExecutionProvider` with `ROCMExecutionProvider` within `mdx.py` on line `69`
